addons/product/models/product_template.py

```diff
@@ -259,10 +257,21 @@ def _get_weight_uom_id_from_ir_config_parameter(self):
         else:
             return self.env.ref('uom.product_uom_kgm')
 
-    def _compute_weight_uom_id(self):
-        weight_uom_id = self._get_weight_uom_id_from_ir_config_parameter()
-        for product_template in self:
-            product_template.weight_uom_id = weight_uom_id
+    @api.depends('weight')
+    def _compute_weight_uom_name(self):
+        for template in self:
+            weight = template._get_weight_uom_id_from_ir_config_parameter()
+            template.weight_uom_name = weight.name
+
+    @api.depends('volume')
+    def _compute_volume_name(self):
+        """ Get the unit of measure to interpret the `volume` field. By default, we consider
+        that volumes are expressed in cubic meters. Users can configure to express them in cubic feet
+        by adding an ir.config_parameter record with "product.product_volume" as key
+        and "1" as value.
+        """
+        get_param = self.env['ir.config_parameter'].sudo().get_param
+        self.volume_name = "ft³" if get_param('product.volume') == '1' else "m³"
```

Comment:
could you do a helper like 'get_weight_uom_id_from_ir_config_parameter' for the volume and use the unit of measure.